src/ui/button/button.tsx

```diff
@@ -0,0 +1,105 @@
+import styles from './button.module.css'
+
+import { MouseEvent, ReactElement } from 'react'
+import cn, { Argument } from 'classnames'
+import { CSSProperties } from 'styled-components'
+
+export interface ButtonProps {
+  text?: string | JSX.Element
+  textSize?: 'xs' | 'sm'
+  size?: 'xs' | 'sm' | 'md'
+  disabled?: boolean
+  variant?:
+    | 'primary'
+    | 'secondary'
+    | 'secondary-dark'
+    | 'tertiary'
+    | 'primary-dark'
+    | 'diestruction'
+  className?: Argument
+  onClick?: (event: MouseEvent<HTMLButtonElement>) => void
+  onMouseUp?: (event: MouseEvent<HTMLButtonElement>) => void
+  onMouseDown?: (event: MouseEvent<HTMLButtonElement>) => void
+  append?: ReactElement
+  prepend?: ReactElement
+  icon?: ReactElement
+  refEl?: any
+  dataTestId?: string
+  style?: CSSProperties
+}
+
+export const Button = ({
+  text,
+  textSize = 'sm',
+  size = 'md',
+  disabled = false,
+  variant = 'primary',
+  onClick,
+  onMouseUp,
+  onMouseDown,
+  className,
+  append,
+  prepend,
+  icon,
+  refEl,
+  dataTestId,
+  style = {},
+}: ButtonProps): ReactElement => {
+  const isSecondary = variant === 'secondary'
```

Comment:
Not needed to create this flags.
You should just use this values in styles direct, something like this:
```
cn(styles.button, styles[`button_${variant}`], styles[`button_${size}`], isOnlyIcon && styles.button_iconOnly)
```

Reply:
Done